solidity/dashboard/src/components/ExplorerModeModal.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from "react"
+import SelectedWalletModal from "./SelectedWalletModal"
+import * as Icons from "./Icons"
+import { WALLETS } from "../constants/constants"
+import { useState } from "react"
+import ExplorerModeAddressForm from "./ExplorerModeAddressForm"
+
+const ExplorerModeModal = ({ connector, connectAppWithWallet, closeModal }) => {
+  const [selectedAddress, setSelectedAddress] = useState("")
+
+  useEffect(() => {
+    connector.setSelectedAccount(selectedAddress)
+  }, [selectedAddress])
+
+  const submitAction = (values) => {
+    setSelectedAddress(values.address)
```

Comment:
Hmmm I think in this case the `ExplorerModeModal` component won't re-render, because we won't be changing the state. So even though we will set the `selectedAddress` for connector in `submitAction` (and set proper condition in return statement) the return statement won't display `<SelectedWalletModal>` component